ddtrace/contrib/wsgi/wsgi.py

```diff
@@ -0,0 +1,159 @@
+
+    return url
+
+
+def get_request_headers(environ):
+    """
+    Manually grab the request headers from the environ dictionary.
+    """
+    request_headers = dict()
+    for key in environ.keys():
+        if key.startswith("HTTP"):
+            request_headers[from_wsgi_header(key)] = environ[key]
+    return request_headers
+
+
+def default_wsgi_span_modifier(span, environ):
+    span.resource = "{} {}".format(environ["REQUEST_METHOD"], environ["PATH_INFO"])
+
+
+class DDTraceWrite(object):
+    def __init__(self, write, tracer):
+        self._write = write
+        self._tracer = tracer
+
+    def __call__(self, data):
+        self._write(data)
+
+
+class DDWSGIMiddleware(object):
+    """WSGI middleware providing tracing around an application.
+
+    :param application: The WSGI application to apply the middleware to.
+    :param tracer: Tracer instance to use the middleware with. Defaults to the global tracer.
+    """
+
+    def __init__(self, application, tracer=None, span_modifier=default_wsgi_span_modifier):
+        self.app = application
+        self.tracer = tracer or ddtrace.tracer
+        self.span_modifier = span_modifier
+
+    def __call__(self, environ, start_response):
+        def intercept_start_response(status, response_headers, exc_info=None):
+            span = self.tracer.current_root_span()
+            status_code, status_msg = status.split(" ", 1)
+            span.set_tag("http.status_msg", status_msg)
+            trace_utils.set_http_meta(span, config.wsgi, status_code=status_code, response_headers=response_headers)
+            with self.tracer.trace(
+                "wsgi.start_response",
+                service=trace_utils.int_service(None, config.wsgi),
+                span_type=SpanTypes.WEB,
+            ):
+                write = start_response(status, response_headers, exc_info)
+            return DDTraceWrite(write, self.tracer)
+
+        if config.wsgi.distributed_tracing:
+            ctx = propagator.extract(environ)
+            if ctx.trace_id:
+                self.tracer.context_provider.activate(ctx)
+
+        with self.tracer.trace(
+            "wsgi.request",
+            service=trace_utils.int_service(None, config.wsgi),
+            span_type=SpanTypes.WEB,
+        ) as span:
+            # This prevents GeneratorExit exceptions from being propagated to the top-level span.
+            # This can occur if a streaming response exits abruptly leading to a broken pipe.
+            # Note: The wsgi.response span will still have the error information included.
+            span._ignore_exception(generatorExit)
+            with self.tracer.trace("wsgi.application"):
+                result = self.app(environ, intercept_start_response)
+
+            with self.tracer.trace("wsgi.response") as resp_span:
+                if hasattr(result, "__class__"):
+                    resp_span.meta["result_class"] = getattr(getattr(result, "__class__", None), "__name__")
```

Comment:
```suggestion
                    resp_class = getattr(getattr(result, "__class__"), "__name__", None)
                    if resp_class:
                        resp_span.meta["result_class"]  = resp_class
```

to avoid the AttributeError if for whatever reason the class does not have a name.